agrona/src/main/java/org/agrona/concurrent/status/AtomicLongPosition.java

```diff
@@ -117,19 +131,27 @@ public void setVolatile(final long value)
      */
     public boolean proposeMax(final long proposedValue)
     {
-        return proposeMaxOrdered(proposedValue);
+        return proposeMaxRelease(proposedValue);
     }
 
     /**
      * {@inheritDoc}
      */
     public boolean proposeMaxOrdered(final long proposedValue)
+    {
+        return proposeMaxRelease(proposedValue);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public boolean proposeMaxRelease(final long proposedValue)
     {
         boolean updated = false;
 
-        if (get() < proposedValue)
+        if (value.get() < proposedValue)
         {
-            setOrdered(proposedValue);
+            value.setRelease(proposedValue);
             updated = true;
         }
```

Comment:
This needs a CAS loop to ensure that update is applied correctly. Similar to `jdk.internal.misc.Unsafe#getAndAddLongRelease`:
```java
    @ForceInline
    public final long getAndAddLongRelease(Object o, long offset, long delta) {
        long v;
        do {
            v = getLong(o, offset);
        } while (!weakCompareAndSetLongRelease(o, offset, v, v + delta));
        return v;
    }
```

Reply:
Why should the concurrency semantics of this method be changed? 

This PR does nothing else than adding a new method with the body of the old, and the old methods pointing to the new.

If we want to change the concurrency semantics, this should be part of a different PR IMHO because there are performance drawbacks. On the X86 it will lead to a lock prefixed instruction which is significantly slower than an X86 loads/store.